src/yarn/typescript-workspace.ts

```diff
@@ -91,18 +133,23 @@ export class TypeScriptWorkspace extends typescript.TypeScriptProject {
     this.monorepo = options.parent;
     this.isPrivatePackage = options.private ?? false;
     this.workspaceDirectory = workspaceDirectory;
+    this.dependencyRange = '^0.0.0';
 
     // If the package is public, all local deps and peer deps must also be public and other TypeScriptWorkspaces
     if (!this.isPrivatePackage) {
-      const illegalDeps = this.workspaceDependencies([DependencyType.RUNTIME, DependencyType.PEER])?.filter(
-        (dep) => 'isPrivatePackage' in dep && dep.isPrivatePackage,
-      );
```

Comment:
What's the actual use case here? My understanding of the code is that taking a dependency on a private package is fine, it just has to be a dev dep. The idea being that on publishing the private package will always be missing.